src/ol_infrastructure/applications/mitxpro/__main__.py

```diff
@@ -128,12 +126,38 @@
 )
 mitxpro_edxapp_db = OLAmazonDB(mitxpro_edxapp_db_config)
 
+hyphenated_db_purpose = (mitxpro_config.require("db_purpose")).replace("_", "-")
+
+edx_role_statments = mysql_sql_statements.update(
+    {
+        f"edxapp-{hyphenated_db_purpose}": {
+            "create": "CREATE USER '{{{{name}}}}'@'%' IDENTIFIED BY '{{{{password}}}}';"
+            "GRANT SELECT, INSERT, UPDATE, DELETE, CREATE, INDEX, DROP, ALTER, REFERENCES"  # noqa: Q000
+            "CREATE TEMPORARY TABLES, LOCK TABLES ON {{{{ app }}}}.* TO '{{{{name}}}}'@'%';",
+            "revoke": "DROP USER '{{{{name}}}}';",
+        },
+        f"edxapp-{hyphenated_db_purpose}": {
+            "create": "CREATE USER '{{{{name}}}}'@'%' IDENTIFIED BY '{{{{password}}}}';"
+            "GRANT SELECT, INSERT, UPDATE, DELETE, CREATE, INDEX, DROP, ALTER, REFERENCES"  # noqa: Q000
+            "CREATE TEMPORARY TABLES, LOCK TABLES ON {{{{ app }}}}.* TO '{{{{name}}}}'@'%';",
+            "revoke": "DROP USER '{{{{name}}}}';",
+        },
+        f"xqueue-{hyphenated_db_purpose}": {
+            "create": "CREATE USER '{{{{name}}}}'@'%' IDENTIFIED BY '{{{{password}}}}';"
+            "GRANT SELECT, INSERT, UPDATE, DELETE, CREATE, INDEX, DROP, ALTER, REFERENCES"  # noqa: Q000
```

Comment:
Should we scope this to the specific DB/schema? e.g. `ON xqueue_{ db_purpose }.*`

Reply:
This would be `xqueue_{ mitxpro_config.db_purpose }.*` given that `db_purpose` is defined as part of the mitxpro_config, right?

Reply:
It would be `xqueue_{mitxpro_config.require('db_purpose')}.*`